pkg/apis/pipeline/v1beta1/task_validation_test.go

```diff
@@ -33,13 +33,6 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-var validResource = v1beta1.TaskResource{
-	ResourceDeclaration: v1beta1.ResourceDeclaration{
-		Name: "validsource",
-		Type: "git",
```

Comment:
technically it would be better to replace this with the image resource as you've done elsewhere (aligns better with our code standards on always being in a releasable state)

Reply:
Thanks for the suggestion, I was also debating this to see what would be a best solution since we are going to be in a stage where all the resourceTypes are removed before we start to remove the generic resourceTypes functions. Does that mean the removal of the last resourceTypes should also include the removal of the generic functions? But until then we need to change them all to the very last resource type?

Reply:
Thanks for the answer offline with Lee.

I am hoping to put this PR as the last removal along with the generic functions of PipelineResources eg. inputresources etc.

Reply:
It might make sense to put a hold on this pr or mark it as a draft in that case